pulser-core/pulser/pulse.py

```diff
@@ -103,10 +103,7 @@ def __init__(
             )
         object.__setattr__(self, "amplitude", amplitude)
         object.__setattr__(self, "detuning", detuning)
-        assert not isinstance(phase, Parametrized)
-        if (phase_ := pm.AbstractArray(phase, dtype=float)).ndim != 0:
-            raise ValueError(f"'phase' must be a single float, not {phase!r}.")
-        object.__setattr__(self, "phase", phase_ % (2 * np.pi))
+        object.__setattr__(self, "phase", cast(float, phase) % (2 * np.pi))
```

Comment:
I think what I had before is what we want here

Reply:
I think phase should be treated on the same footing as amplitude and detuning. From the perspective of pulser-diff, user might do something like `phase=torch.rand(1, requires_grad=True)` and pass this tensor to pulse constructor. However, the phase here will be a 1D tensor and your implementation will fail. I suggest then changing your version to `phase_ := pm.AbstractArray(phase, dtype=float)).size != 1`. This will handle my case.

Reply:
Yes, your example has `ndim=0` because you create a tensor from a single float. Mine in the comment above has actually `ndim=1` because that's the way `rand` is implemented. At least in my local version of `torch` :)